src/ape/utils/process.py

```diff
@@ -0,0 +1,49 @@
+import queue
+import threading
+import time
+
+from ape.exceptions import SubprocessTimeoutError
+
+
+class JoinableQueue(queue.Queue):
+    """
+    A queue that can be joined, useful for multi-processing.
+    Borrowed from the ``py-geth`` library.
+    """
+
+    def __iter__(self):
+        while True:
+            item = self.get()
+
+            is_stop_iteration_type = isinstance(item, type) and issubclass(item, StopIteration)
+            if isinstance(item, StopIteration) or is_stop_iteration_type:
+                return
+
+            elif isinstance(item, Exception):
+                raise item
+
+            elif isinstance(item, type) and issubclass(item, Exception):
+                raise item
+
+            yield item
+
+    def join(self, timeout=None):
+        with SubprocessTimeoutError(timeout) as _timeout:
+            while not self.empty():
+                time.sleep(0)
+                _timeout.check()
+
+
+def spawn(target, *args, **kwargs):
```

Comment:
As long as we're aware that this is still running on the same processor, just being dropped off the main thread, I'm okay with this. I tend to avoid multi-threading in Python, and use multi-process. And considering this is more of a CPU bound issue, maybe multiprocessing is the move here? I'll thread if it's I/O bound